### ISSUE TITLE
Occasional failure in xrdp-keygen

### ISSUE THREAD

**Issue by proski** [neutrinolabs/xrdp#331]
Once in a while, `xrdp-keygen` fails to generate the key file. The error can be reproduced by running `xrdp-keygen` in a loop for a minute or two:

```
while :; do
  ./xrdp-keygen xrdp rsakeys.ini || exit 1
done
```

It turns out this check fails in `common/ssl_calls.c` on line 505:

    error = len != pri_len;

pri_len is 64, but len can become 63. As I understand it, some random number can become so small that it fits 63 bytes instead of 64. Does it need to be an error? If yes, can `xrdp-keygen` retry the call?

The RPM package I maintain generates the key at the install time. The failure means that the package fails to install. That's quite bad.

**Comment by jsorg71** [comment]
I see, by random, the top byte can be zero and the length of the modulus is one bytes less.  I guess it could be 2 or more too but less likely.

**Comment by jsorg71** [comment]
should be fixed in 0d192aee62c6eab93611f2dde088711d7764bacd

**Comment by proski** [comment]
Thank you! I ran the test for 30 minutes, no failures have been detected. Resolving.
